engine.py

```diff
@@ -170,9 +201,37 @@ def post_app_init(self):
         # We have some mode-specific initialization to do.
         if self.hiero_enabled:
             self.post_app_init_hiero(menu_name)
+        elif self.studio_enabled:
+            self.post_app_init_studio(menu_name)
         else:
             self.post_app_init_nuke(menu_name)
 
+    def post_app_init_studio(self, menu_name="Shotgun"):
+        """
+        The Nuke Studio specific portion of the engine's post-init process.
+        """
+        if self.has_ui:
+            # Note! not using the import as this confuses Nuke's callback system
+            # (several of the key scene callbacks are in the main init file).
+            import tk_nuke
+
+            # Create the menu!
+            self._menu_generator = tk_nuke.NukeStudioMenuGenerator(self, menu_name)
+            self._menu_generator.create_menu()
+
+            import hiero
+            def _set_project_root_callback(event):
```

Comment:
Not a fan of method defined within methods, even less within if's. Lambda's are great for this. Your call.